platform/newsfeed/src/lib.rs

```diff
@@ -0,0 +1,135 @@
+use serde::{Deserialize, Serialize};
+use std::env;
+use tracing::error;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum ClaudeModel {
+    #[serde(rename = "claude-3-5-sonnet-20240620")]
+    Claude35Sonnet,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum Role {
+    #[serde(rename = "user")]
+    User,
+    #[serde(rename = "system")]
+    System,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Prompt {
+    pub role: Role,
+    pub content: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ClaudeMessage {
+    pub model: ClaudeModel,
+    pub max_tokens: u64,
+    pub system: Option<String>,
+    pub tool_choice: Option<serde_json::Value>,
+    pub tools: Vec<serde_json::Value>,
+    pub messages: Vec<Prompt>,
+}
+
+impl ClaudeMessage {
+    pub async fn generate(&self) -> Result<String, reqwest::Error> {
+        let client = reqwest::Client::new();
+
+        let claude_api_key = match env::var("CLAUDE_API_KEY") {
+            Ok(result) => result,
+            Err(e) => {
+                error!("Failed to get ANTHROPIC_API_KEY: {:?}", e);
+                "".to_string()
+            }
+        };
+
+        let response = client
+            .post("https://api.anthropic.com/v1/messages")
+            .header("Content-Type", "application/json")
+            .header("anthropic-version", "2023-06-01")
+            .header("x-api-key", claude_api_key)
+            .body(serde_json::to_string(self).unwrap())
+            .send()
+            .await?
+            .text()
+            .await?;
+
+        Ok(response)
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct SentimentResponse {
+    pub confidence: u64,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct ClaudeResponseContent {
+    #[serde(rename = "type")]
+    type_: String,
+    id: String,
+    name: String,
+    pub input: SentimentResponse,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct ClaudeResponseUsage {
+    input_tokens: u64,
+    output_tokens: u64,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct ClaudeResponse {
+    pub id: String,
+    #[serde(rename = "type")]
+    pub type_: String,
+    pub role: String,
+    pub model: String,
+    pub content: Vec<ClaudeResponseContent>,
+    pub stop_reason: String,
+    pub stop_sequence: Option<String>,
+    pub usage: ClaudeResponseUsage,
+}
+
+#[cfg(test)]
```

Comment:
Is it Rust's pattern to include the tests in the code file? If not, could we split it out?

Reply:
yep